webui/module/Api/config/module.config.php

```diff
@@ -0,0 +1,253 @@
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/timeline[/][:id]',
+                            'constraints' => array(
+                                'id' => '[a-zA-Z0-9][a-zA-Z0-9\._-]*',
+                            ),
+                            'defaults' => array(
+                                'controller' => 'Api\Controller\Timeline',
+                            ),
+                        ),
+                    ),
+                    'execute' => array(
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/execute[/][:id]',
+                            'constraints' => array(
+                                'id' => '[a-zA-Z0-9][a-zA-Z0-9\._-]*',
+                            ),
+                            'defaults' => array(
+                                'controller' => 'Api\Controller\ExecuteOnDir',
+                            ),
+                        ),
+                    ),
+                    'client' => array(
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/client[/][:id]',
+                            'constraints' => array(
+                                'id' => '[a-zA-Z0-9][a-zA-Z0-9\._-]*',
+                            ),
+                            'defaults' => array(
+                                'controller' => 'Api\Controller\Client',
+                            ),
+                        ),
+                    ),
+                    'fileset' => array(
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/fileset[/][:id]',
+                            'constraints' => array(
+                                'id' => '[a-zA-Z0-9][a-zA-Z0-9\._-]*',
+                            ),
+                            'defaults' => array(
+                                'controller' => 'Api\Controller\Fileset',
+                            ),
+                        ),
+                    ),
+                    'schedule' => array(
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/schedule[/][:id]',
+                            'constraints' => array(
+                                'id' => '[a-zA-Z0-9][a-zA-Z0-9\._-]*',
+                            ),
+                            'defaults' => array(
+                                'controller' => 'Api\Controller\Schedule',
+                            ),
+                        ),
+                    ),
+                    'storage' => array(
+                        'type' => Segment::class,
+                        'options' => array(
+                            'route' => '/storage[/][:id]',
```

Comment:
Currently we use the storage name as an identifier via query parameter, because the storage name may contain blanks which causes trouble in routing. However, as the storage ID is now available via API 2, wasn't always the case, this could be changed.